runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/ErrorMappingSourceTypeReferencesExist.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal.validation;
+
+import static java.lang.String.format;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+import static org.mule.runtime.ast.api.util.ComponentAstPredicatesFactory.currentElemement;
+import static org.mule.runtime.ast.api.validation.Validation.Level.ERROR;
+import static org.mule.runtime.extension.api.ExtensionConstants.ERROR_MAPPINGS_PARAMETER_NAME;
+
+import org.mule.runtime.api.component.ComponentIdentifier;
+import org.mule.runtime.api.message.ErrorType;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.api.ComponentAst;
+import org.mule.runtime.ast.api.ComponentParameterAst;
+import org.mule.runtime.extension.api.error.ErrorMapping;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+/**
+ * Referenced error types do exist in the context of the artifact
+ */
+public class ErrorMappingSourceTypeReferencesExist extends AbstractErrorTypesValidation {
```

Comment:
This class is quite similar to `ErrorHandlerOnErrorTypeExists`.  Why not extract the common code to a superclass?